src/core/annotation.js

```diff
@@ -2083,6 +2083,12 @@ class ButtonWidgetAnnotation extends WidgetAnnotation {
       return;
     }
 
+    if (this.appearance) {
+      // See https://bugzilla.mozilla.org/show_bug.cgi?id=1722036
+      // If we've an AS and V then take AS.
+      this.data.fieldValue = this._decodeFormValue(params.dict.get("AS"));
```

Comment:
I'd suggest reducing any *possible* regression risk here, given all the weird things that PDF generators does, by checking that the /AS value makes sense before using it; e.g. something along these lines:
```javascript
      const asValue = this._decodeFormValue(params.dict.get("AS"));
      if (typeof asValue === "string") {
        this.data.fieldValue = asValue;
      }
```